### PR TITLE
.

### PR DESCRIPTION
**Pull request by PaltineanuRares956**

### CODE REVIEW

File: Paltineanu Rares-Mihai/3.5 - curs5/main.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        } else {
+            colD = 0;
+            rowsD++;
+        }
+
+    }
+
+}
+
+int main() {
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.5 - curs5/main.c:100]: Great work modularizing everything and splitting it into functions!

I don't know if you want this, but, there's also a way to allocate the matrix in a function as well, as to not have repeated code. I don't have a compiler at hand but you should be able to write something like this:

```
void allocateMatrix(int *** matrix, int rows, int colums) {
    *matrix = (int **) malloc(sizeof(int*) * rows);
    for(int i = 0; i < rows; i++) {
        *(matrix[i]) = (int *) malloc(sizeof(int) * colums);
    }
}
```

and then you should be able to call that function from your main file:
```
...
int ** A, B, C, D;
allocateMatrix(&A, n, n);
allocateMatrix(&B, n, n);
allocateMatrix(&C, n, n);
allocateMatrix(&D, n, n);
```

Pretty sure it should work, but maybe I'm missing something - tell me if it does or doesn't!

File: Paltineanu Rares-Mihai/3.7 - curs5/main.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/**Given three positive integers, representing a year, a month and a day-of-the-month respectively,
+write a function to compute the number of the day in the year, and the number of days to the end of that year
+*/
+#include <stdio.h>
+#include <stdlib.h>
+int months[] = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.7 - curs5/main.c:6]: Yes, good way of representing the number of days for each month

File: Paltineanu Rares-Mihai/3.7 - curs5/main.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+void printfdate(int m, int d) {
+
+    switch(d)
+    {
+        case 1: printf("The number of the 1st of "); break;
+        case 2: printf("The number of the 2nd of"); break;
+        case 3: printf("The number of the 3rd of"); break;
+        default: printf("The number of the %dth of", d);
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.7 - curs5/main.c:25]: I might be nitpicky, but you are missing `21st, 22nd, 23rd and 31st`  😄

File: Paltineanu Rares-Mihai/3.8 - curs5/main.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            }
+            else {
+                nr += currentValue;
+            }
+        } else {
+            nr += romanLetters(Roman[i + 1]) - currentValue;
+            i++;
+        }
+    }
+    if(lastLetter == 1)
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.8 - curs5/main.c:52]: Curly brackets

File: Paltineanu Rares-Mihai/3.8 - curs5/main.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        nr += romanLetters(Roman[size - 1]);
+    return nr;
+}
+
+int main() {
+
+    char Roman[50];
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.8 - curs5/main.c:59]: Code looks fine, rules look OK too (can't really tell if you caught all cases but you might have)
Great!

File: Paltineanu Rares-Mihai/3.9 - curs5/main.c [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+char roman1[3][3] = {"M", "MM", "MMM"};
+char roman2[9][5] = {"C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"};
+char roman3[9][5] = {"X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"};
+char roman4[9][5] = {"I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"};
+int conversion(int nr, char *roman) {
+
+    int arab, size = 0;
+
+    arab = nr / 1000;
+    if(arab != 0)
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.9 - curs5/main.c:15]: Curly brackets everywhere!

**Review by Alex-Cosma — COMMENTED** [review]
I really like how clean you keep your `main` function, this is really great and you should keep this as a principle!

File: Paltineanu Rares-Mihai/3.10 - curs5/main.c [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        (*ct)++;
+        printf("%d  ", money);
+    }
+
+}
+
+int main() {
+
+    int RON, ct = 0; scanf("%d", &RON);
+    banknotes(&RON, &ct, 500);
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.10 - curs5/main.c:20]: As to not have so many repeated function calls, you could have written an array which would contain all these values, e.g. `int noteValues = [500, 200, 100, 50, 10 ,5, 1];`
and then with a for loop, simply iterate through them and calling the `banknotes` function for each value

File: Paltineanu Rares-Mihai/3.13 - curs5/main.c [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    int sizePerfect = 0;
+    read(vect);
+
+    for(int i = 0; i < n; i++) {
+        if(square(vect[i])) {
+            perfect[sizePerfect++] = vect[i];
+        }
+    }
+    if(sizePerfect != 0){
+    write(perfect, sizePerfect);
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.13 - curs5/main.c:46]: `format use AStyle`

File: Paltineanu Rares-Mihai/3.4 - curs5/main.c [hunk 9]
```diff
@@ -0,0 +1,105 @@
+/**Two character strings representing very large decimal integers are given.
+Write a program to perform arithmetic operations with such very large decimal integers
+*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+///It's only the addition, in this weekend maybe i will do the subtraction and multiplication
+///I just want to know if what i did is ok
```
Comment (Alex-Cosma) [inline Paltineanu Rares-Mihai/3.4 - curs5/main.c:9]: What you did is good, you're on the right track. Good job - I'm curious if you'll be able to reuse this function when dealing with addition

**Comment by Alex-Cosma** [comment]
Awesome job, I'm going to merge this for now but feel free to work on it more and push new content when you have the chance

File: Paltineanu Rares-Mihai/3.5 - curs5/main.c (hunk 1 above)
Comment (PaltineanuRares956) [reply Paltineanu Rares-Mihai/3.5 - curs5/main.c:100]: it's not working

File: Paltineanu Rares-Mihai/3.7 - curs5/main.c (hunk 3 above)
Comment (PaltineanuRares956) [reply Paltineanu Rares-Mihai/3.7 - curs5/main.c:25]: I forgot about these cases :)))

File: Paltineanu Rares-Mihai/3.9 - curs5/main.c (hunk 6 above)
Comment (PaltineanuRares956) [reply Paltineanu Rares-Mihai/3.9 - curs5/main.c:15]: what?

**Review by Alex-Cosma — COMMENTED** [review]
Curly brackets = `{`
e.g. whenever you have 
```
if (arab != 0)
    code ...
```

you should replace it with

``` 
if (arab != 0) {
    code ...
}
```

notice the extra `{` and `}`

**Comment by Alex-Cosma** [comment]
Not sure why that thing isn't working, will check at lab

File: Paltineanu Rares-Mihai/3.4 - curs5/main.c (hunk 9 above)
Comment (PaltineanuRares956) [reply Paltineanu Rares-Mihai/3.4 - curs5/main.c:9]: You mean multiplication?

Comment (Alex-Cosma) [reply Paltineanu Rares-Mihai/3.4 - curs5/main.c:9]: Oups, yes, that's what I meant

**Pull request merged**